packaging/packaging.go

```diff
@@ -0,0 +1,293 @@
+
+		return tarFileName
+
+	}
+	return ""
+}
+
+func WritePart(logger logr.Logger, fileName string, csvReader *csv.Reader, num int) (string, bool) {
+	log := logger.WithValues("costmanagement", "WritePart")
+	fileNamePart := strings.TrimSuffix(fileName, ".csv")
+	sizeEstimate := 0
+	splitFileName := fileNamePart + strconv.Itoa(num) + ".csv"
+	log.Info("Creating file ", "file", splitFileName)
+	splitFile, err := os.Create(splitFileName)
+	if err != nil {
+		fmt.Println("An error occurred:", err)
+	}
+	// Create the csv writer
+	writer := csv.NewWriter(splitFile)
+	// err = writer.WriteRow(csvHeader)
+	for {
+		row, err := csvReader.Read()
+		if err == io.EOF {
+			return splitFileName, true
+		}
+		writer.Write(row)
+		rowLen := len(strings.Join(row, ","))
+		rowSize := rowLen + int(float64(rowLen)*variance)
+		sizeEstimate += rowSize
+		if sizeEstimate >= int(maxBytes) {
+			return splitFileName, false
+		}
+	}
+}
+
+func SplitFiles(logger logr.Logger, filePath string) {
+	fileList, err := ioutil.ReadDir(filePath)
+	if err != nil {
+		fmt.Println("could not read dir")
+	}
+	for _, file := range fileList {
+		absPath := filePath + "/" + file.Name()
+		info, err := os.Stat(absPath)
+		if err != nil {
+			fmt.Println("error: ", err)
+		}
+		fileSize := info.Size()
+		if fileSize >= maxBytes {
+			var splitFiles []string
+			// var csvHeader string
+			// open the file
+			csvFile, err := os.Open(absPath)
+			if err != nil {
+				fmt.Println("An error occurred ::", err)
+			}
+			csvReader := csv.NewReader(csvFile)
+			// csvHeader := next(csvReader)
+			var part int64 = 1
+			for {
+				newFile, eof := WritePart(logger, absPath, csvReader, int(part))
+				splitFiles = append(splitFiles, newFile)
+				part += 1
+				if eof || part >= maxSplits {
+					break
+				}
+			}
+			os.Remove(absPath)
+			fmt.Println(splitFiles)
+		}
+	}
+}
+
+func Split(logger logr.Logger, filePath string, cost *costmgmtv1alpha1.CostManagement) {
+	log := logger.WithValues("costmanagement", "Split")
+	var outFiles []string
+	log.Info("Checking to see if the report files need to be split")
+	needSplit := NeedSplit(filePath)
+	if needSplit {
+		log.Info("Report files exceed the max size. Splitting files")
+		SplitFiles(logger, filePath)
+		tarPath := filePath + "/../"
+		tarFileTmpl := "cost-mgmt"
+		fileList := BuildLocalCSVFileList(filePath)
+		manifestFileName, manifestUUID := RenderManifest(logger, fileList, cost, filePath)
+		for idx, fileName := range fileList {
+			if strings.Contains(fileName, ".csv") {
+				var singleFile []string
+				singleFile = append(singleFile, fileName)
+				tarFileName := tarPath + tarFileTmpl + strconv.Itoa(idx) + ".tar.gz"
+				log.Info("Generating tar.gz", "tarFile", tarFileName)
+				outputTar := WriteTarball(logger, tarFileName, manifestFileName, manifestUUID, singleFile)
+				if outputTar != "" {
+					outFiles = append(outFiles, outputTar)
+				}
+			}
+		}
+	} else {
+		tarFileName := filePath + "/../cost-mgmt.tar.gz"
```

Comment:
Same as above, can we use absolute paths?

Reply:
done